File: musify_flutter/lib/src/features/playlists/domain/playlist.dart
class PlayList {
  const PlayList({
    required this.id,
    required this.name,
    required this.description,
    required this.smallImageKeyName,
    required this.mediumImageKeyName,
    required this.largeImageKeyName,
    required this.createdAt,
    required this.updatedAt,
  });

  final String id;
  final String name;
  final String description;
  final String smallImageKeyName;
  final String mediumImageKeyName;
  final String largeImageKeyName;
  final DateTime createdAt;
  final DateTime updatedAt;

  factory PlayList.fromJson(Map<String, dynamic> json) {
    return PlayList(
      id: json['id'] as String,
      name: json['name'] as String? ?? 'Sin nombre',
      description: json['description'] as String? ?? '',
      smallImageKeyName: json['smallImageKeyName'] as String? ?? '',
      mediumImageKeyName: json['mediumImageKeyName'] as String? ?? '',
      largeImageKeyName: json['largeImageKeyName'] as String? ?? '',
      createdAt: DateTime.parse(json['createdAt'] as String),
      updatedAt: DateTime.parse(json['updatedAt'] as String),
    );
  }
}

File: backend/Musify.Domain/Entities/PlayList.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

using Musify.Domain.Abstractions;
using Musify.Domain.ValueObjects;

namespace Musify.Domain.Entities
{
#pragma warning disable CS8618
    [Table("PlayLists")]
    public class PlayList : IAuditable
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        public required long UserId { get; set; }

        [Required]
        [MaxLength(50)]
        public required string Name { get; set; }

        [Required]
        [MaxLength(50)]
        public required string NormalizedName { get; set; }

        [Required]
        [MaxLength(256)]
        public string? Description { get; set; }

        public PlayListPictures? Pictures { get; set; }

        [Required]
        public LifeCycleStatus LifeCycleStatus { get; set; } = LifeCycleStatus.Active;

        [Required]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Required]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        [ForeignKey(nameof(UserId)) ]
        public User User { get; set; }

        public ICollection<PlayListHasTrack> PlayListTracks { get; set; } = new List<PlayListHasTrack>();
    }
}